Lickability/PinpointKit
# Swift 3.0 Xcode 8 GM Migration

## What It Does

Opens the workspace in Xcode 8 beta 6:

- Fixed compiler errors (59a0fbe), including making only things that _need_ to be `fileprivate` `fileprivate`, instead of what the migrator prefers, which is much more lenient converting `private` → `fileprivate`.
- Ran the migrator (558d0af and fd08856), rejecting all `private` → `fileprivate` changes.

## How to Test

- Open the project in Xcode 8 beta 6, build, and smoke test.
- Review the changes in each commit to ensure you agree with my and the migrator’s choices.

## Notes

- This is based on the branches from #157 and #159, so please review / merge those first. There’s only one added commit here (790688a). Alternatively, we could close #157 and #159, but the separation will be easier to review the changes in isolation. I’ll leave that up to the reviewer.
- This is pointed at `swift-3.0`, not `develop`.
- Information regarding `open`: https://github.com/apple/swift-evolution/blob/master/proposals/0117-non-public-subclassable-by-default.md#open-class-members
- Information regarding `fileprivate`: https://github.com/apple/swift-evolution/blob/7fcba970b88a5de3d302d291dc7bc9dfba0f9399/proposals/0025-scoped-access-level.md
- See also [this blog post for a helpful summary](https://swifting.io/blog/2016/08/17/22-swift-3-access-control-beta-6/) that discusses `open` and `fileprivate`.
